src/renderers/dom/fiber/wrappers/ReactDOMFiberInput.js

```diff
@@ -258,11 +257,13 @@ var ReactDOMInput = {
       case 'week':
         // This fixes the no-show issue on iOS Safari and Android Chrome:
         // https://github.com/facebook/react/issues/7233
-        node.value = '';
-        node.value = node.defaultValue;
+        //
+        // Important: use setAttribute instead of node.type = "x" to avoid
+        // an exception in IE10/11 due to an unrecognized input type
+        node.setAttribute('type', 'text');
```

Comment:
I was under the impression that you can't change the type of an input after it's created in at least some versions of IE. Do you mind double checking which to be sure we don't break something?

Reply:
This is definitely true for some browsers for password types, for "security".

Reply:
I made a quick test:
https://codepen.io/nhunzaker/full/BwPbRK/

This works in:

- IE 9, 10, 11
- Firefox 47
- Chrome 41
- Safari Desktop 7.0
- Safari iOS 9

Is this a reasonable test case? This page is over HTTPs. I wonder if it would be different otherwise. 

In any case, this only applies to date/time/color inputs, and only when they are mounted for the first time.

Reply:
Ahh, sorry. Key detail! This **fails** in IE8, but not IE9 or higher:

<img width="1466" alt="screen shot 2017-10-12 at 5 50 48 pm" src="https://user-images.githubusercontent.com/590904/31521001-e6411622-af75-11e7-94aa-b569a94c1121.png">

Reply:
Thanks! I think this should be OK then.

Long-term I suppose someone could make an IE8-compatible renderer using the react-reconciler package if they wanted to.